crates/router/src/core/payment_link/payment_link_initiate/payment_link.js

```diff
@@ -619,7 +619,7 @@ function renderDynamicMerchantDetails(paymentDetails) {
 function appendMerchantDetails(paymentDetails, merchantDynamicDetails) {
   if (
     !(
-      typeof paymentDetails.transaction_details === "object" &&
+      Array.isArray(paymentDetails.transaction_details) &&
```

Comment:
Wouldn't this return `false` every time? 

```suggestion
  !(
      !Array.isArray(paymentDetails.transaction_details)
        && paymentDetails.transaction_details !== null
        && typeof paymentDetails.transaction_details === "object"
        && Object.keys(paymentDetails.transaction_details).length > 0
   )
```

Reply:
it returns true only if the transaction_details is an array otherwise even for null and object it would throw false